integrations/google_vertex/src/haystack_integrations/components/generators/google_vertex/chat/gemini.py

```diff
@@ -106,6 +124,17 @@ def _generation_config_to_dict(self, config: Union[GenerationConfig, Dict[str, A
             "stop_sequences": config._raw_generation_config.stop_sequences,
         }
 
+    def _tool_config_to_dict(self, tool_config: ToolConfig) -> Dict[str, Any]:
```

Comment:
Since the newly added `_tool_config_to_dict` and `_tool_config_from_dict `are similar in both `VertexAIGeminiGenerator` and `VertexAIGeminiChatGenerator`, one option is to create a helper class to centralize these functions and reduce code duplication. Alternatively, we can leave the current structure as is.